sklearn/gaussian_process/tests/test_gpr.py

```diff
@@ -652,3 +652,27 @@ def test_gpr_predict_error():
     err_msg = "At most one of return_std or return_cov can be requested."
     with pytest.raises(RuntimeError, match=err_msg):
         gpr.predict(X, return_cov=True, return_std=True)
+
+
+def test_y_std_with_multitarget_normalized():
+    """
+    Regression test for issues #17394 and #18065.
+    Check if GPR can compute y_std in predict() method when normalize_y==True
+    in multi-target regression.
+    """
+    X_train = np.random.rand((11, 10))
+    # 6 target features -> multi-target
+    y_train = np.random.rand((11, 6))
+    X_test = np.random.rand((4, 10))
+
+    # Generic kernel
+    kernel = kernels.ConstantKernel(1.0, (1e-1, 1e3))
+    kernel *= kernels.RBF(10.0, (1e-3, 1e3))
+
+    # normalize_y == True
+    model = GaussianProcessRegressor(kernel=kernel,
+                                     n_restarts_optimizer=10,
+                                     alpha=0.1,
+                                     normalize_y=True)
+    model.fit(X_train, y_train)
+    y_pred, std = model.predict(X_test, return_std=True)
```

Comment:
```suggestion
    y_pred, y_std = model.predict(X_test, return_std=True)
```

Reply:
It is true that the previous code was failing already. But we could make a couple of assertions regarding the shape of the returned values.

```python
assert y_pred.shape == (n_samples, n_targets)
```

and something similar for `y_std`